src/compiler/utilities.ts

```diff
@@ -3563,6 +3566,15 @@ export function getJSDocCommentsAndTags(hostNode: Node, noCache?: boolean): read
             result = addRange(result, (noCache ? getJSDocTypeParameterTagsNoCache : getJSDocTypeParameterTags)(node as TypeParameterDeclaration));
             break;
         }
+        if (isBindingElement(node) && isIdentifier(node.name)) {
+            const root = getRootDeclaration(node);

```

Comment:
please extract this function, putting it next to the ones for parameters and type parameters.